### ISSUE TITLE
ScreenIdle FeatureState

### ISSUE THREAD

**Issue by erichs** [Netflix-Skunkworks/stethoscope-app#60]
**Note: for support questions, please [email us](mailto:stethoscope@netflix.com) or contact us on [Gitter](https://gitter.im/Netflix-Stethoscope/Lobby)**. This repository's issues are reserved for feature requests and bug reports.

* **I'm submitting a ...**
  - [ ] bug report
  - [x] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

* **What is the motivation / use case for changing the behavior?**

Many compliance frameworks have requirements around **idle lockout**, where the screensaver locks after some duration.

* **Please tell us about your environment:**

  - Stethoscope Version: 2.0.5
  - Platform:
    - [x] Mac
    - [x] Windows
    - [x] Linux
  - Browser: [all]

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

On Mac, this may be implemented almost identically to the existing `screenLock` resolver, with the exception that the `idleTime` field is available via `osquery` even on High Sierra.

In terms of schema, this might be:
```
screenIdle: String
```

and configured perhaps like:

```
...
screenLock: IF_SUPPORTED
screenIdle: "120"
...
```

osquery results for this field less than or equal to the policy would PASS. Perhaps a semver-like "<=120" syntax could be supported.

**Comment by erichs** [comment]
For platform completeness' sake, I'll note that on Windows 10, this setting is retrievable at the following registry key: `HKEY_CURRENT_USER\Control Panel\Desktop\ScreenSaveTimeout`, and for Ubuntu linux, it is available (on the default window manager used by 16.04) by shelling out to: `gsettings get org.gnome.desktop.screensaver ubuntu-lock-on-suspend`.

**Comment by rmcvey** [comment]
Hey @erichs, thanks for filing. I neglected to document this feature, but it is currently available (for Windows) as a policy argument:

Here is the default (value is in seconds):
```json
{
    windowsMaxScreenLockTimeout: 600
}
```

Here is the relevant code (pardon the commented out bits, I've been trying to dial this setting in). https://github.com/Netflix-Skunkworks/stethoscope-app/blob/master/resolvers/platform/WindowsSecurity.js#L50-L68

Would be great to add support for Ubuntu. I'll see what I can do there.

**Comment by erichs** [comment]
Nifty! Thanks @rmcvey. 

FWIW, I took a stab at spiking a `screenIdle` resolver for Mac:  https://github.com/erichs/stethoscope-app/commit/f1304a89f430f0a81e50b11a9d5076d378683c9f

This *almost* JustWorked ™️, but I ran into an issue where the args passed into the Mac platform resolver were `{}` at the time `screenIdle` is called, so the comparison with the policy setting (undefined at that point) always returned false. Maybe I missed something obvious.
